docs/examples/shading/plot_martinez_shade_loss.py

```diff
@@ -0,0 +1,276 @@
+"""
+Modelling shading losses in modules with bypass diodes
+======================================================
+"""
+
+# %%
+# This example illustrates how to use the model proposed by Martinez et al.
+# [1]_. The model proposes a power output losses factor by adjusting
+# the incident direct and circumsolar beam irradiance fraction of a PV module
+# based on the number of shaded *blocks*. A *block* is defined as a group of
+# cells protected by a bypass diode. More information on *blocks* can be found
+# in the original paper [1]_ and in the
+# :py:func:`pvlib.shading.direct_martinez` documentation.
+#
+# The following key functions are used in this example:
+#
+# 1. :py:func:`pvlib.shading.direct_martinez` to calculate the power output
+#    losses fraction due to shading.
+# 2. :py:func:`pvlib.shading.shaded_fraction1d` to calculate the fraction of
+#    shaded surface and consequently the number of shaded *blocks* due to
+#    row-to-row shading.
+# 3. :py:func:`pvlib.tracking.singleaxis` to calculate the rotation angle of
+#    the trackers.
+#
+# .. sectionauthor:: Echedey Luis <echelual (at) gmail.com>
+#
+# Problem description
+# -------------------
+# Let's consider a PV system with the following characteristics:
+#
+# - Two north-south single-axis trackers, each one having 6 modules.
+# - The rows have the same true-tracking tilt angles. True tracking
+#   is chosen in this example, so shading is significant.
+# - Terrain slope is 7 degrees downward to the east.
+# - Rows' axes are horizontal.
+# - The modules are comprised of multiple cells. We will compare these cases:
+#    - modules with one bypass diode
+#    - modules with three bypass diodes
+#    - half-cut cell modules with three bypass diodes on portrait and landscape
+#
+# Setting up the system
+# ----------------------
+# Let's start by defining the system characteristics, location and the time
+# range for the analysis.
+
+import pvlib
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.dates import ConciseDateFormatter
+
+pitch = 4  # meters
+width = 1.5  # meters
+gcr = width / pitch  # ground coverage ratio
+N_modules_per_row = 6
+axis_azimuth = 180  # N-S axis
+axis_tilt = 0  # flat because the axis is perpendicular to the slope
+cross_axis_tilt = -7  # 7 degrees downward to the east
+
+latitude, longitude = 40.2712, -3.7277
+locus = pvlib.location.Location(
+    latitude,
+    longitude,
+    tz="Europe/Madrid",
+    altitude=pvlib.location.lookup_altitude(latitude, longitude),
+)
+
+times = pd.date_range("2001-04-11T04", "2001-04-11T20", freq="10min")
+
+# %%
+# True-tracking algorithm and shaded fraction
+# -------------------------------------------
+# Since this model is about row-to-row shading, we will use the true-tracking
+# algorithm to calculate the trackers rotation. Back-tracking reduces the
+# shading between rows, but since this example is about shading, we will not
+# use it.
```

Comment:
```suggestion
# algorithm to calculate the trackers rotation. Back-tracking eliminates
# shading between rows, and since this example is about shading, we will not
# use it.
```